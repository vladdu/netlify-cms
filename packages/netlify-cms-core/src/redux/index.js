import { createStore, applyMiddleware, compose } from 'redux';
import thunkMiddleware from 'redux-thunk';
import waitUntilAction from './middleware/waitUntilAction';
import reducer from 'Reducers/combinedReducer';

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunkMiddleware, waitUntilAction),
    window.__REDUX_DEVTOOLS_EXTENSION__ ? window.__REDUX_DEVTOOLS_EXTENSION__() : f => f,
  ),
);

export default store;
