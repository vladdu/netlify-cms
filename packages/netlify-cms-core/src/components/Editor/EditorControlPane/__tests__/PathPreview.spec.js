import React from 'react';
import { Map } from 'immutable';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import ConnectedPathPreview, { PathPreview } from '../PathPreview';

jest.mock('Reducers', () => {
  return () => ({});
});
jest.mock('Selectors/entryDraft');
jest.mock('react-polyglot', () => {
  return {
    translate: () => Component => props => <Component {...props} t={jest.fn(key => key)} />,
  };
});

describe('PathPreview', () => {
  it('should render successfully and match snapshot', () => {
    const props = {
      value: 'posts/2019/index.md',
      t: jest.fn(key => key),
    };
    const { asFragment } = render(<PathPreview {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });
});

function renderWithRedux(ui, { initialState, store = createStore(() => ({}), initialState) } = {}) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}

describe('ConnectedPathPreview', () => {
  const { selectDraftPath } = require('Selectors/entryDraft');

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should use existing path when newRecord is false', () => {
    const props = {
      collection: Map({
        name: 'posts',
      }),
      entry: Map({
        newRecord: false,
        data: {},
        path: 'existing-path/index.md',
      }),
    };
    const { asFragment, getByTestId } = renderWithRedux(<ConnectedPathPreview {...props} />);

    expect(selectDraftPath).toHaveBeenCalledTimes(0);

    expect(getByTestId('site_path-preview')).toHaveTextContent('existing-path/index.md');

    expect(asFragment()).toMatchSnapshot();
  });

  it('should evaluate preview path when newRecord is true', () => {
    selectDraftPath.mockReturnValue('preview-path/index.md');
    const props = {
      collection: Map({
        name: 'posts',
      }),
      entry: Map({
        newRecord: true,
        data: {},
      }),
    };
    const { asFragment, getByTestId } = renderWithRedux(<ConnectedPathPreview {...props} />);

    expect(selectDraftPath).toHaveBeenCalledTimes(1);
    expect(selectDraftPath).toHaveBeenCalledWith({}, props.collection, props.entry);

    expect(getByTestId('site_path-preview')).toHaveTextContent('preview-path/index.md');

    expect(asFragment()).toMatchSnapshot();
  });
});
