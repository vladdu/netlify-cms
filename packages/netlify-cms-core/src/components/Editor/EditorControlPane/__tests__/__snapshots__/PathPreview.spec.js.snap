// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedPathPreview should evaluate preview path when newRecord is true 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: #798291;
  display: inline-block;
  font-size: 12px;
  font-weight: 600;
  border: 0;
  padding-left: 5px;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  position: relative;
}

<label
    class="emotion-0 emotion-1"
    data-testid="site_path-preview"
  >
    preview-path/index.md
  </label>
</DocumentFragment>
`;

exports[`ConnectedPathPreview should use existing path when newRecord is false 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: #798291;
  display: inline-block;
  font-size: 12px;
  font-weight: 600;
  border: 0;
  padding-left: 5px;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  position: relative;
}

<label
    class="emotion-0 emotion-1"
    data-testid="site_path-preview"
  >
    existing-path/index.md
  </label>
</DocumentFragment>
`;

exports[`PathPreview should render successfully and match snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: #798291;
  display: inline-block;
  font-size: 12px;
  font-weight: 600;
  border: 0;
  padding-left: 5px;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  position: relative;
}

<label
    class="emotion-0 emotion-1"
    data-testid="site_path-preview"
  >
    posts/2019/index.md
  </label>
</DocumentFragment>
`;
